cirq-google/cirq_google/engine/virtual_engine_factory.py

```diff
@@ -197,18 +233,18 @@ def create_noiseless_virtual_engine_from_templates(
     if len(processor_ids) != len(template_names):
         raise ValueError('Must provide equal numbers of processor ids and template names.')
 
-    specifications = []
-    for idx in range(len(processor_ids)):
-        path = pathlib.Path(__file__).parent.parent.resolve()
-        f = open(path.joinpath('devices', 'specifications', template_names[idx]))
-        proto_txt = f.read()
-        f.close()
-        device_spec = v2.device_pb2.DeviceSpecification()
-        text_format.Parse(proto_txt, device_spec)
-        specifications.append(device_spec)
+    specifications = [
+        _create_device_spec_from_template(template_name) for template_name in template_names
+    ]
     return create_noiseless_virtual_engine_from_proto(processor_ids, specifications, gate_sets)
 
 
+MOST_RECENT_TEMPLATES = {
```

Comment:
Should we move this to the top of the file for visibility?

Reply:
done